### PR TITLE
Update confusing handler function name in BasicModal cmp

### PR DESCRIPTION
**Pull request by patchwork-body**
**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

**Summary:**
Use more relative name for `keyup` event handler function name

`@avaya-dux/dux-devs`: event handler name doesn't reflect it's purpose

- Before: `onKeyDown` event handler used to handle `keyup` event 💩
- Now: `onKeyUp` event handler used to handle `keyup` event 🦄

## Summary by CodeRabbit

- **Bug Fixes**
	- Updated keyboard event handling in the BasicModal component to improve accessibility and user interaction by switching from "keydown" to "keyup" events.

### CODE REVIEW

**Comment by patchwork-body** [comment]
@avaya-dux/dux-devs

**Review by joe-s-avaya — APPROVED** [review]

**Review by yangAtSpoken — CHANGES_REQUESTED** [review]
For handling keyboard events in React, `onKeyDown` is generally the best choice due to its immediate feedback and ability to handle key combinations effectively.  Thus I am rejecting this PR.

File: src/components/Modal/BasicModal/BasicModal.tsx [hunk 1]
```diff
@@ -57,11 +57,11 @@ export const BasicModal = forwardRef(
 		);
 
 		useEffect(() => {
-			document.addEventListener("keyup", onKeyDown, false);
+			document.addEventListener("keyup", onKeyUp, false);
 			return () => {
-				document.removeEventListener("keyup", onKeyDown, false);
+				document.removeEventListener("keyup", onKeyUp, false);
```
Comment (joe-s-avaya) [inline src/components/Modal/BasicModal/BasicModal.tsx:62]: @yangAtSpoken, I'm responding to [your comment](https://github.com/avaya-dux/neo-react-library/pull/548#pullrequestreview-2334088662). He's not changing functionality, he's only changing our name to match what the event listener is doing.

Comment (yangAtSpoken) [reply src/components/Modal/BasicModal/BasicModal.tsx:62]: I noticed we have a few places using keyUp. So I will let this go through.

**Review by yangAtSpoken — APPROVED** [review]

**Pull request merged**